examples/hotrod/pkg/log/spanlogger.go

```diff
@@ -16,15 +16,244 @@
 package log
 
 import (
+	"fmt"
 	"time"
 
 	"github.com/opentracing/opentracing-go"
 	tag "github.com/opentracing/opentracing-go/ext"
 	"github.com/opentracing/opentracing-go/log"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/trace"
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
 )
 
+type otelSpanLogger struct {
+	logger     *zap.Logger
+	span       trace.Span
+	spanFields []zapcore.Field
+}
+
+func (sl otelSpanLogger) Debug(msg string, fields ...zapcore.Field) {
```

Comment:
These methods do not directly depend on underlying span so we could keep a single spanLogger type and only differentiate lower in the implementation which span it is dealing with.